### ISSUE TITLE
chapter1---softmax_regression.py，损失函数定义有问题

### ISSUE THREAD

**Issue by ican73** [hzy46/Deep-Learning-21-Examples#17]
chapter1---softmax_regression.py，line29~30损失函数定义有问题

# 根据y, yhat构造交叉熵损失
# cross_entropy = tf.reduce_mean(-tf.reduce_sum(y * tf.log(yhat)))
cross_entropy = tf.reduce_mean(-tf.reduce_sum(y * tf.log(yhat),axis=1))

**Comment by lxsyz** [comment]
参看 #7

**Comment by ican73** [comment]
抱歉，没看到之前的issue，这里确定应该加上，否则会和使用的mini-batch size产生关系，且最终计算的cost function是总和而非均值。
